rbac/common/sawtooth/batcher.py

```diff
@@ -264,27 +265,41 @@ def dict_to_protobuf(message, message_name, dictionary):
 def message_to_message(
     message_to, message_from, message_name=None, exclude_fields=None
 ):
-    """Makes a message from another message
+    """Makes a message from another message or dictionary
     Shallow copy; supports primitive data types and lists (arrays)
     Add depth and/or additional datatypes when needed (TODO: add json support)"""
     if not message_name:
         message_name = type(message_to)
-    for key, value in message_from.ListFields():
-        if hasattr(message_to, key.name) and key.name not in exclude_fields:
-            try:
-                if isinstance(value, list):
-                    getattr(message_to, key.name).extend(value)
-                else:
-                    setattr(message_to, key.name, value)
-            except Exception:
-                raise AttributeError(
-                    "Unable to set attribute {} on {} to value {}".format(
-                        key.name, message_name, value
+    if not exclude_fields:
+        exclude_fields = []
+    if isinstance(message_from, dict):
+        for key, value in message_from.items():
+            if hasattr(message_to, key) and key not in exclude_fields:
```

Comment:
The ignored pylint warning here is indicating too much conditional logic. If you assign key/key.name, then all that follows is duplicate code that could go into its own function.

Reply:
resolved